src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/CSharpSyntaxFormattingService.cs

```diff
@@ -63,4 +50,16 @@ protected override IFormattingResult CreateAggregatedFormattingResult(SyntaxNode
         protected override AbstractFormattingResult Format(SyntaxNode node, SyntaxFormattingOptions options, IEnumerable<AbstractFormattingRule> formattingRules, SyntaxToken startToken, SyntaxToken endToken, CancellationToken cancellationToken)
             => new CSharpFormatEngine(node, options, formattingRules, startToken, endToken).Format(cancellationToken);
     }
+
+#if !CODE_STYLE
+    [ExportLanguageService(typeof(ISyntaxFormattingService), LanguageNames.CSharp), Shared]
```

Comment:
I can't decide if this is better than the old code - was there a motivation behind switching to a separate class rather than removing the exports/instance?

Reply:
yes.  a core concept here is that MEF services can only be referenced by using mef importing, while the code-style helpers are only referenced by direct class reference.  These two approaches are incompatible within the same type.

Reply:
Handled in: https://github.com/dotnet/roslyn/pull/60462